challenge_14/action_controller.rb

```diff
@@ -23,6 +24,44 @@ class Blog < ActiveRecord::Base; end
   end
 end
 
+class AppController < ActionController::Base
+  def root
+    response = ""
+    response += "<p><strong>Submit a new Blog Post!</p></strong>"
+    response += "<form method='post' enctype='application/x-www-form-urlencoded' action='/create-post'>"
+    response += "<p><label>Blog Title: <input name='title'></label></p>"
+    response += "<p><label>Content: <textarea name='content'></textarea></label></p>"
+    response += "<p><button>Submit post</button></p>"
+    response += "</form>"
+    render html: response.html_safe
+  end
+
+  def show_data
+    response = ""
+    response += "<ul>"
+
+    Blog.all.each do |blog|
+      response += "<li>"
+      response += "<strong>Title: #{CGI.escape_html(blog.title)}</strong>, Content: #{CGI.escape_html(blog.content)}"
+      response += "</li>"
+    end
+    response += "</ul>"
+    render html: response.html_safe
+  end
+
+  def create_post
+    puts 'Got a new POST request!'
+
+    Blog.create!(params.permit(:title, :content))
```

Comment:
As we discussed when pairing, this could also be `Blog.create!(title: params[:title], content: params[:content])`, because that effectively achieves the same goal of preventing any unintended parameters from slipping through. `#permit` is tidier though!